scipy/stats/tests/test_sampling.py

```diff
@@ -582,5 +597,230 @@ def test_bad_urn_factor(self):
+                             zip(dists, mvs))
+    def test_basic(self, dist, mv_ex):
+        rng = NumericalInversePolynomial(dist, random_state=42)
+        check_cont_samples(rng, dist, mv_ex)
+
+    very_slow_dists = ['studentized_range', 'trapezoid', 'triang', 'vonmises',
+                       'levy_stable', 'kappa4', 'ksone', 'kstwo', 'levy_l',
+                       'gausshyper', 'anglit']
+    # for some reason, UNU.RAN segmentation faults for the uniform.
+    fatal_fail_dists = ['uniform']
+    # fails for unbounded PDFs
+    unbounded_pdf_fail_dists = ['arcsine', 'beta']
+    # for these distributions, some assertions fail due to minor
+    # numerical differences. They can be avoided either by changing
+    # the seed or by increasing the u_resolution.
+    fail_dists = ['ncf', 'pareto', 'chi2', 'fatiguelife', 'halfgennorm',
+                  'gilbrat', 'lognorm', 'ncx2', 't']
+
+    @pytest.mark.xslow
+    @pytest.mark.parametrize("distname, params", distcont)
+    def test_basic_all_scipy_dists(self, distname, params):
+        if distname in self.very_slow_dists:
+            pytest.skip(f"PINV too slow for {distname}")
+        if distname in self.fail_dists:
+            pytest.skip(f"PINV fails for {distname}")
+        if distname in self.unbounded_pdf_fail_dists:
+            pytest.skip("PINV fails for unbounded PDFs.")
+        if distname in self.fatal_fail_dists:
+            pytest.xfail(f"PINV segmentation faults for {distname}")
+        dist = (getattr(stats, distname)
+                if isinstance(distname, str)
+                else distname)
+        dist = dist(*params)
+        with suppress_warnings() as sup:
+            sup.filter(RuntimeWarning)
+            rng = NumericalInversePolynomial(dist, random_state=42)
+        check_cont_samples(rng, dist, [dist.mean(), dist.var()])
+
+    @pytest.mark.parametrize("pdf, err, msg", bad_pdfs_common)
+    def test_bad_pdf(self, pdf, err, msg):
+        class dist:
+            pass
+        dist.pdf = pdf
+        with pytest.raises(err, match=msg):
+            NumericalInversePolynomial(dist, domain=[0, 5])
+
+    # test domains with inf + nan in them. need to write a custom test for
+    # this because not all methods support infinite tails.
+    @pytest.mark.parametrize("domain, err, msg", inf_nan_domains)
+    def test_inf_nan_domains(self, domain, err, msg):
+        with pytest.raises(err, match=msg):
+            NumericalInversePolynomial(StandardNormal(), domain=domain)
+
+    u = [np.linspace(0, 1, num=10000), [], [[]], [np.nan],
+         [-np.inf, np.nan, np.inf], 0,
+         [[np.nan, 0.5, 0.1], [0.2, 0.4, np.inf], [-2, 3, 4]]]
+
+    @pytest.mark.parametrize("u", u)
+    def test_ppf(self, u):
+        dist = StandardNormal()
+        rng = NumericalInversePolynomial(dist, u_resolution=1e-14)
+        # Older versions of NumPy throw RuntimeWarnings for comparisons
+        # with nan.
+        with suppress_warnings() as sup:
```

Comment:
this might be a bit confusing. I guess the idea is to check the following scenarios:

- for finite values, use all_close to compare the finite values
- for special cases like values outside of [0,1], I guess the result should be nan (?)
- ppf(0) = -inf, ppf(1) = inf
- nans and empty values map to nan (?)

maybe add a brief comment inline or split up the test?

Reply:
> maybe add a brief comment inline or split up the test?

Adding an inline comment for the purpose of each test case sounds good. I will do that.

Reply:
Done.